rollup/pipeline/pipeline.go

```diff
@@ -266,6 +266,8 @@ func (p *Pipeline) traceAndApplyStage(txsIn <-chan *types.Transaction) (<-chan e
 					p.nextL1MsgIndex = tx.AsL1MessageTx().QueueIndex + 1
 				}
 
+				p.state.IntermediateRoot(p.chain.Config().IsEIP158(p.Header.Number))
```

Comment:
Is there any overhead when we call this after each transaction, as opposed to once per block? (related historical nodt: https://github.com/ethereum/EIPs/issues/98)

Reply:
Yeah, overall, calculating state commitment per txn is more costly. But we are trying to utilize a time period which is otherwise completely wasted. We can remove this once CCC is no longer the bottleneck.

Reply:
Ok, sounds good. Let's add a comment about this for future context.

Reply:
done https://github.com/scroll-tech/go-ethereum/pull/802/commits/25b93b83c2b4fb8ed385b824a138a7171a58e586